File: engine/oapi.php
<?php
include 'dbworks.php';//KillerTheGame - table

function authOpenAPIMember()
{ //vk copypaste
  include 'privatedata.php';

  $session = array(); 
  $member = FALSE; 
  $valid_keys = array('expire', 'mid', 'secret', 'sid', 'sig'); 
  $app_cookie = $_COOKIE['vk_app_5170996']; //app id
  if ($app_cookie) { 
    $session_data = explode ('&', $app_cookie, 10); 
    foreach ($session_data as $pair) { 
      list($key, $value) = explode('=', $pair, 2); 
      if (empty($key) || empty($value) || !in_array($key, $valid_keys)) { 
        continue; 
      } 
      $session[$key] = $value; 
    } 
    foreach ($valid_keys as $key) { 
      if (!isset($session[$key])) return $member; 
    } 
    ksort($session); 

	  $sign = ''; 
    foreach ($session as $key => $value) { 
      if ($key != 'sig') { 
        $sign .= ($key.'='.$value); 
      } 
    } 
    $sign .= $vksecretkey; //sc key
    $sign = md5($sign);

	  //mine
	  $playerinfo = mysql_num_rows(mysql_query("SELECT * FROM KillerTheGame WHERE vk_id = '".$session['mid']."'"));
	  if ($session['sig'] == $sign && $session['expire'] > time()) { 
      $member = array( 
        'id' => intval($session['mid']), 
        'secret' => $session['secret'], 
        'sid' => $session['sid'],
		    'player' => (($playerinfo === false || $playerinfo === 0) ? false : true)
      ); 
    } 
  }
  
  return $member; 
} 
?>

File: hsespionage/oapi.py
from hsespionage import privatedata
from hsespionage import postgresdb
from flask import request
from string import Template
import collections
import hashlib
import time

def authOpenAPIMember():
    session = {}
    member = {}
    valid_keys = ("expire", "mid", "secret", "sid", "sig")
    app_cookie = request.cookies.get("vk_app_5170996")
    if app_cookie:
        session_data = app_cookie.split("&", maxsplit=10)

        for pair in session_data:
            key, value = pair.split("=", maxsplit=2)
            if (not key) or (not value) or (key not in valid_keys):
                continue
            session[key] = value

        for key in valid_keys:
            if key not in session:
                return member

        session = collections.OrderedDict(sorted(session.items()))

        sign = ""
        for key, value in session.items():
            if key != "sig":
                sign += (key + "=" + value)
        sign += privatedata.vksecretkey
        sign = hashlib.md5(sign)

        if session["sig"] == sign and session["expire"] > time.time():
            member = {
                "id": int(session["mid"]),
                "secret": session["secret"],
                "sid": session["sid"],
                "player": (True if postgresdb.one(Template("SELECT vk_id FROM hsspies_game WHERE vk_id='$vk_id'").substitute(
                    vk_id=postgresdb.string.escape_literal(session["mid"]))) else False)
            }

    return member
    